Python3 + minor fixes

ayushnawal opened the pull request:
PR includes:
* port to python 3
* update sugargame to v1.3
* add license and repository metadata
* add readme

ayushnawal committed:
add license and repository metadata

ayushnawal committed:
add readme

ayushnawal committed:
update sugargame to v1.3

ayushnawal committed:
Port to Python 3

ayushnawal commented:
@quozl please review!

quozl commented:
Tested.  Doesn't work.

```log
Traceback (most recent call last):
  File "/usr/bin/sugar-activity3", line 5, in <module>
    activityinstance.main()
  File "/usr/lib/python3.7/dist-packages/sugar3/activity/activityinstance.py", line 179, in main
    module = __import__(module_name)
  File "/usr/share/sugar/activities/Adivinanzas.activity/AdivinanzasActivity.py", line 17, in <module>
    import sugargame.canvas
  File "/usr/share/sugar/activities/Adivinanzas.activity/sugargame/canvas.py", line 7, in <module>
    import event
ModuleNotFoundError: No module named 'event'

```

ayushnawal commented:
Can you please suggest me the steps to reproduce this error?

I didn't encounter the same while testing.

Can you please check the latest commit(especially the sugargame commit) because i replaced `import event` with `import sugargame.event as event`

quozl commented:
Thanks.  Tested 6242e84.  Reviewed.

* [ ] the activity is not part of the Sugar desktop,
* [ ] the link to activities.sugarlabs.org is not useful for Python 3 activities.

ayushnawal committed:
update readme

ayushnawal commented:
done

ayushnawal committed:
update readme

ayushnawal commented:
@quozl please review

quozl commented on activity/activity.info:
```diff
@@ -1,6 +1,8 @@
 icon = activity-generic
 activity_version = 3
+repository = https://github.com/sugarlabs/adivinanzas-activity.git
+license = GPLv3+
```
Why did you put _license_ here?

ayushnawal replied:
Added license because of the warning displayed in logs while running the activity.

I am removing it now, there may be some copyright issues of the original author here, I remember you said about copyright and license in a pull request in some other activity previously.

If there is nothing like that, I will commit that again later.

quozl commented on adivinanzas.py:
```diff
@@ -7,8 +7,7 @@ def leer(nombre_archivo):
     # Ahora tenemos todo el contenido del archivo en la variable "cuerpo".
-    cuerpo = unicode(archivo.read())
-    cuerpo.decode("utf-8")
+    cuerpo = str(archivo.read())
```
Why is `str` necessary?  Doesn't open with mode `r` guarantee str?

ayushnawal replied:
Didn't see the mode `r`, sorry.

You are absolutely right, str is not at all necessary here.

In Python 3 when we use in text mode `r`, Python will parse the file according to the text encoding you give it (or if you don't give one, a platform-dependent default), and `read()` will give you an `str`.

Since python3, The default encoding for Python source code is UTF-8 (also mentioned in the `python 3` documentation)

quozl commented on adivinanzas.py:
```diff
@@ -7,8 +7,7 @@ def leer(nombre_archivo):
-    cuerpo = unicode(archivo.read())
-    cuerpo.decode("utf-8")
+    cuerpo = str(archivo.read())
     # unicode() transforma el string en un string con soporte de acentos
```
Why leave outdated comments?

ayushnawal replied:
removing these comments, sorry for that.

Didn't notice comments while reviewing changes after porting

quozl reviewed (changes requested):
Thanks.  Reviewed.

ayushnawal committed:
Changes made as requested

remove outdated comments
remove str because using r mode and read() will give str
remove license

quozl commented:
Thanks for the changes.  They look fine now.

In reply to your comments about licensing, I've researched briefly.

Regarding the license of Adivinanzas;
* the text file [creditos.txt](https://raw.githubusercontent.com/sugarlabs/adivinanzas-activity/master/creditos.txt) does contain a clear claim of authorship by a group of people, and a license which translates to "_Creative Commons License Attribution Share Alike 3.0 and +_", with the first two authors @icarito and @kaametza, who were members of Sugar Labs,
* you and I are not the authors, so we cannot change the license or copyright without contacting the authors,

My opinion;
* until instructed otherwise, the license is as it is stated,
* that the toolkit [warns that the license is unspecified](https://github.com/sugarlabs/sugar-toolkit-gtk3/blob/master/src/sugar3/bundle/activitybundle.py#L226)  must not be taken as permission to try to change it against the wishes of the authors,
* while the Creative Commons licenses are not used for software in Anglophile countries because of how copyright law evolved, [Can I apply a Creative Commons license to software?](https://creativecommons.org/faq/#can-i-apply-a-creative-commons-license-to-software) 
"_We recommend against using Creative Commons licenses for software._", that doesn't mean we can't maintain the activity, as long as you are happy that your changes are also licensed under Creative Commons.

p.s. please don't call me _Sir_, it is disrespect in my culture.
